### DISCUSSION TITLE
Is it possible to name the official architecture guidance

### DISCUSSION THREAD

**Discussion by madroidmaq** [android/nowinandroid#546] [General]
I don't know if it's an appropriate place to ask this question, but I'd be happy to ask if there is a better channel. my question is:

**Is it possible to name the [official architecture guidance](https://developer.android.com/topic/architecture)？**

Everyone knows what we're talking about when we mention the name, [VIPER](https://www.objc.io/issues/13-architecture/viper/), [Clean Architecture](https://blog.cleancoder.com/uncle-bob/2012/08/13/the-clean-architecture.html), [Jetpack Compose](https://developer.android.com/jetpack/compose), [Hilt](https://developer.android.com/jetpack/androidx/releases/hilt), etc.

I personally encounter the following scenarios:

1.  case1: I often see some inaccurate publicity in some technical communities, such as: "Learn the official Android architecture: MVI", "Let's learn the official Android MVVM architecture" and similar titles. I know this statement is not rigorous, because apart from some architectural components, the Android architecture specification does not have a definite name, so it is not MVI or other architectures.
2. case2: When I want to discuss and exchange this part of the content with my colleagues, I need to describe clearly what architecture I am talking about: for example, it adopts a data-driven and one-way data flow method, which is generally divided into three layers of UI Layer , Domain Layer, Data Layer, etc. Although these need to be described as clearly as possible at the beginning, when we communicate this information frequently, a short name can improve the efficiency and accuracy of communication. I don't think keywords like MVVM can replace the new Android App Architecture Guidelines.

I see the name "Modern Android App Architecture" used in [Learning pathway](https://developer.android.com/courses/pathways/android-architecture), but obviously this name is too long, I tried to use the acronym "MAAA", which is obviously too crude. Some good names that come to mind:
- **[VIPER](https://www.objc.io/issues/13-architecture/viper/)**, he is the abbreviation of each keyword: is a backronym for **View**, **Interactor**, **Presenter**, **Entity**, and **Routing** .
- **[Hilt](https://developer.android.com/jetpack/androidx/releases/hilt)**: This is a dependency injection library that encapsulates Dagger with clear semantics.

I also tried to name this set of excellent architecture guidelines myself, but I found that I didn't find a suitable name. As stated in the document, the current architectural approach is not only applicable to the Android platform, but also applicable to the iOS and Flutter platforms, so I think this set of naming should accommodate these platforms as much as possible.

To sum up, I think the Android official architecture guide should be named, so that the architecture guide can **be disseminated more easily and accurately**.

**Comment by JoseAlcerreca** [comment]
Thanks for taking time to post this. This question was already answered on a YouTube comment and we saw it in the other channels. However, it's not related to NiA.

**Reply by madroidmaq** [reply]
Thank you for your reply, I would like to use this platform to see if there are people who think the same as me, if not, I will close this discussion.
